src/app/g3w-ol/controls/basequerypolygoncontrol.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * @file
+ * @since v3.8
+ */
+import { SPATIALMETHODS, VM } from 'g3w-ol/constants';
+import GUI from 'services/gui';
+
+const { merge }                  = require('core/utils/ol');
+const InteractionControl         = require('g3w-ol/controls/interactioncontrol');
+
+// TODO: make it easier to understand.. (what variables are declared? which ones are aliased?)
+const {
+  Geometry : {
+    getAllPolygonGeometryTypes,
+    isPolygonGeometryType
+  }
+} = require('core/utils/geo');
+
+const VALIDGEOMETRIES = getAllPolygonGeometryTypes();
+
+const BaseQueryPolygonControl = function(options = {}) {
+
+  const {
+    spatialMethod=SPATIALMETHODS[0],
+    name,
+    tipLabel,
+    label,
+    onSelectlayer,
+    enabled=true,
+    interactionClass
+  } = options;
```

Comment:
Check if rewritten like so could be equivalent:

```suggestion
const BaseQueryPolygonControl = function(options = {
    spatialMethod = SPATIALMETHODS[0],
    name,
    tipLabel,
    label,
    onSelectlayer,
    enabled = true,
    interactionClass
  } = {}) {

```

Reply:
@Raruto that's wrong because:

` const _options = {
    offline: false,
    name,
    tipLabel,
    label,
    // update selected layer
    onSelectlayer,
    clickmap: true, // set ClickMap
    interactionClass,
    spatialMethod,
    toggledTool:{
      type: 'spatialMethod',
      how: 'toggled' // or hover
    },
    onhover: true
  };`

and for example tipLabel is not defined.

I rewrite it